File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    
    # _list[크기] = 개수
    # 큰거 부터 차례로
    
    _dict = {}
    
    for size in tangerine:
        _dict[size] = _dict.get(size, 0) + 1
    
    _list = sorted(list(_dict.values()))
    
    for i in _list[::-1]:
        if k <= 0:
            return answer
        # print(k, i)
        k -= i
        answer += 1
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;

        Map<Integer, Integer> gyul = new HashMap<>();
        
        for(int size :  tangerine) {
            gyul.put(size, gyul.getOrDefault(size, 0) + 1);
        }

        List<Integer> valList = gyul.values().stream().sorted().collect(Collectors.toList());
        
//         for(int val: valList) {
//             System.out.println(val);

//         }
        
        int index = valList.size() -1;
        // System.out.println("---");
        while(k >= 0) {
            // System.out.println(k);
            k -= valList.get(index);
            index -= 1;
            answer += 1;
            
            if(k<= 0) {
                break;
            }
        }
        
        return answer;
    }
}